### ISSUE TITLE
KanjiPad:   java.lang.IllegalArgumentException  in java.util.AbstractList.subList

### ISSUE THREAD

**Issue by mvysny** [mvysny/aedict#707]
From the Google Play Console:
```
java.lang.RuntimeException: java.lang.IllegalArgumentException
	at sk.baka.aedictkanjipadext.QueryService$IncomingHandler$1.onFailure(QueryService.java:37)
	at sk.baka.aedictkanjipadext.util.LoaderEx$Loader.onFailure(LoaderEx.java:168)
	at sk.baka.aedictkanjipadext.util.LoaderEx$Loader.access$100(LoaderEx.java:96)
	at sk.baka.aedictkanjipadext.util.LoaderEx$Loader$1.run(LoaderEx.java:145)
	at sk.baka.aedictkanjipadext.App$2.run(App.java:91)
	at android.os.Handler.handleCallback(Handler.java:815)
	at android.os.Handler.dispatchMessage(Handler.java:104)
	at android.os.Looper.loop(Looper.java:207)
	at android.app.ActivityThread.main(ActivityThread.java:5710)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:900)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:761)
Caused by: java.lang.IllegalArgumentException
	at java.util.AbstractList.subList(AbstractList.java:736)
	at com.leafdigital.kanji.KanjiList.getTopMatches(KanjiList.java:237)
	at sk.baka.aedictkanjipadext.Searcher$MyLoader.load(Searcher.java:163)
	at sk.baka.aedictkanjipadext.Searcher$MyLoader.loadItemsInBackground(Searcher.java:139)
	at sk.baka.aedictkanjipadext.util.LoaderEx$Loader.run(LoaderEx.java:116)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:423)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
	at java.lang.Thread.run(Thread.java:818)
```

**Comment by mvysny** [comment]
Need more info

**Comment by mvysny** [comment]
```
java.lang.RuntimeException: java.lang.RuntimeException: start>end 36 32 32 4 10 9
	at sk.baka.aedictkanjipadext.QueryService$IncomingHandler$1.onFailure(QueryService.java:37)
	at sk.baka.aedictkanjipadext.util.LoaderEx$Loader.onFailure(LoaderEx.java:232)
	at sk.baka.aedictkanjipadext.util.LoaderEx$Loader.access$100(LoaderEx.java:149)
	at sk.baka.aedictkanjipadext.util.LoaderEx$Loader$1.run(LoaderEx.java:208)
	at android.os.Handler.handleCallback(Handler.java:815)
	at android.os.Handler.dispatchMessage(Handler.java:104)
	at android.os.Looper.loop(Looper.java:207)
	at android.app.ActivityThread.main(ActivityThread.java:5710)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:900)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:761)
Caused by: java.lang.RuntimeException: start>end 36 32 32 4 10 9
	at com.leafdigital.kanji.KanjiList.getTopMatches(KanjiList.java:243)
	at sk.baka.aedictkanjipadext.Searcher$KanjiMatchingLoader.load(Searcher.java:177)
	at sk.baka.aedictkanjipadext.Searcher$KanjiMatchingLoader.loadItemsInBackground(Searcher.java:153)
	at sk.baka.aedictkanjipadext.util.LoaderEx$Loader.run(LoaderEx.java:179)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
	at java.lang.Thread.run(Thread.java:818)
```

**Comment by mvysny** [comment]
The problem was that on a powerful devices (with 10 cores) the kanji match tasks were incorrectly sliced. Fixed in 1.6.3
